src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -0,0 +1,133 @@
+
+/**
+ * Structural Variant caller
+ *
+ * This currently looks for regions with abnormally large insert sizes. This typically indicates
+ * that there's been a large deletion.
+ *
+ * It does this by finding the average insert size of all mate pairs which overlap each 25-base pair "block" of the
+ * genome. If this average is significantly different than the average insert size genome-wide, then it's evidence that
+ * there may be a structural variant.
+ * 
+ * The output is a text file of genomic ranges where there's some evidence of an SV.
+ */
+object StructuralVariant {
+
+  val MAX_INSERT_SIZE = 25000
+  val BLOCK_SIZE = 25
+
+  protected class Arguments extends Common.Arguments.Reads with DistributedUtil.Arguments {
+    @Args4JOption(name = "--filter-contig", usage = "Filter to alignments where either mate is in this contig.")
+    var filterContig: String = ""
+
+    @Args4JOption(name = "--output", usage = "Path to output text file.")
+    var output: String = ""
+  }
+
+  case class Location(
+      contig: String,
+      position: Long) {
+  }
+
+  case class GenomeRange(
+    contig: String,
+    start: Long,
+    stop: Long)
+
+  object Caller extends SparkCommand[Arguments] {
+    override val name = "structural-variant"
+    override val description = "Find structural variants, i.e. large insertions or deletions, inversions and translocations."
+
+    // The sign of inferredInsertSize follows the orientation of the read. This returns
+    // an insert size which is positive in the common case, regardless of the orientation.
+    def orientedInsertSize(r: PairedRead[MappedRead]): Option[Int] = {
+      val sgn = if (r.isPositiveStrand) { 1 } else { -1 }
+      r.mateAlignmentProperties.flatMap(_.inferredInsertSize).map(x => x * sgn)
+    }
+
+    // Coalesce sequences of values separated by blockSize into single ranges.
+    // The sequence of values must be sorted.
+    def coalesceAdjacent(xs: IndexedSeq[Long], blockSize: Long): Iterator[(Long, Long)] = {
+      val rangeStarts = ArrayBuffer(0)
+      for (i <- 1 until xs.length) {
+        if (xs(i - 1) + blockSize != xs(i)) {
+          rangeStarts += i
+        }
+      }
+      rangeStarts += xs.length
+
+      rangeStarts.sliding(2).map(items => {
+        val startIdx = items(0)
+        val nextStartIdx = items(1)
+        (xs(startIdx), xs(nextStartIdx - 1))
+      })
+    }
+
+    // Returns a sequence of all blocks which overlap the read, its mate or the insert between them.
+    def matedReadBlocks(r: PairedRead[MappedRead]): Seq[Long] = {
+      if (!r.isMateMapped) {
+        Seq[Long]()
+      } else {
+        val mateStart = r.mateAlignmentProperties.map(mp => mp.start).getOrElse(r.read.start)
+        val start = List(r.read.start, r.read.end, mateStart).min
+        val roundedStart = (1.0 * start / BLOCK_SIZE).toLong * BLOCK_SIZE
+        val insertSize = orientedInsertSize(r).getOrElse(0)
+        (roundedStart to (roundedStart + insertSize) by BLOCK_SIZE)
```

Comment:
intellij gives me an "unnecessary parends" warning here and on L117, if we care about those